packages/dds/matrix/src/test/time/sharedMatrix.spec.ts

```diff
@@ -0,0 +1,134 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+import { benchmark, isInPerformanceTestingMode } from "@fluid-tools/benchmark";
+
+import { SharedMatrix } from "../../index.js";
+import { createLocalMatrix } from "../utils.js";
+
+describe("SharedMatrix execution time", () => {
+	// The value to be set in the cells of the matrix.
+	const matrixValue = "cellValue";
+	// The test matrix's size will be 10*10, 100*100, 1000*1000.
+	const matrixSizes = isInPerformanceTestingMode
+		? [10, 100, 1000]
+		: // When not measuring perf, use a single smaller data size so the tests run faster.
+			[10];
+
+	// The number of operations to perform on the matrix.
+	const operationCounts = isInPerformanceTestingMode
+		? [10, 100, 1000]
+		: // When not measuring perf, use a single smaller data size so the tests run faster.
+			[5];
+
+	let localMatrix: SharedMatrix;
+	for (const matrixSize of matrixSizes) {
+		describe(`Size of ${matrixSize}*${matrixSize} SharedMatrix`, () => {
+			// Filter counts to ensure remove operation do not exceed matrixSize
+			const validRemoveCounts = operationCounts.filter((count) => count <= matrixSize);
+
+			// Insert-related tests that are not limited by matrixSize
+			for (const count of operationCounts) {
+				beforeEach(() => {
+					localMatrix = createLocalMatrix({
```

Comment:
I'm still catching up with the matrix performance test (so I might be wrong), but isn't this bit *unfair* for set/remove test?  

For the insert-related tests, we create the matrix *beforehand* by calling the `createLocalMatrix()` in `beforeEach`, while in the set/remove-related tests we pass the same function into the `benchmarkFn()` thus adding additional noise in the execution time. 

On my codespace, the average remove/set operation at `1000*1000` matrix is just `+0.08` second than the insert operation so the difference is infinitesimal but still wanted to check if this is a valid concern.

Reply:
This actually made me realize there's _almost_ a subtle bug here. Since we're defining `beforeEach()` inside a for loop, we're really creating three `beforeEach()` functions inside the same suite, and that means all three are running for every test in the suite. Luckily, in this case the three are doing the exact same thing because they don't depend on the loop variable (`count`), but if we were able to use `beforeEach()`, it should have gone outside the loop.